core/src/repair/repair_service.rs

```diff
@@ -122,6 +122,36 @@ impl RepairStatsGroup {
     }
 }
 
+#[derive(Debug)]
+pub struct RepairMetrics {
+    pub stats: RepairStats,
+    pub best_repairs_stats: BestRepairsStats,
+    pub timing: RepairTiming,
+    pub last_report: Instant,
+}
+
+impl Default for RepairMetrics {
+    fn default() -> Self {
+        Self {
+            stats: RepairStats::default(),
+            best_repairs_stats: BestRepairsStats::default(),
+            timing: RepairTiming::default(),
+            last_report: Instant::now(),
+        }
+    }
+}
+
+impl RepairMetrics {
+    pub fn maybe_report(&mut self) {
+        if self.last_report.elapsed().as_secs() > 2 {
```

Comment:
I don't like this style since it's easy to think we report every 2 seconds but it's actually 3... but decided to not change it with this PR